auth-foundation/src/main/java/com/okta/authfoundation/credential/CredentialDataSource.kt

```diff
@@ -15,187 +15,85 @@
     ): Credential {
         val storageIdentifier = UUID.randomUUID().toString()
         val idToken = token.idToken?.let { jwtParser.parse(it) }
         val credential =
-            Credential(oidcClient, this, storageIdentifier, token, tags)
+            Credential(storageIdentifier, token, tags = tags)
         credentialsCache[storageIdentifier] = credential
         storage.add(
             token,
             Token.Metadata(
                 storageIdentifier,
                 tags,
-                idToken?.deserializeClaims(JsonObject.serializer()),
-                isDefault
+                idToken
             ),
             security
         )
-        oidcClient.configuration.eventCoordinator.sendEvent(CredentialCreatedEvent(credential))
+        AuthFoundationDefaults.eventCoordinator.sendEvent(CredentialCreatedEvent(credential))
         return credential
     }
 
-    /**
-     * Replaces the stored values of [Credential] associated with [id], and returns the [Credential] with updated values.
-     *
-     * @param id The id of the stored [Credential].
-     * @param token The token object to store.
-     * @param tags The new tags for replacing the stored tags of this [Credential]. If null, keep the previous stored tags.
-     * @param security The new [Credential.Security] level for this token. If null, keep the previous stored [Credential.Security] level.
-     * @param isDefault Specify whether the [Credential] with [id] should be set as default. If null, the default [Credential] is unchanged.
-     */
-    suspend fun replaceCredential(
-        id: String,
-        token: Token,
-        tags: Map<String, String>? = null,
-        security: Credential.Security? = null,
-        isDefault: Boolean? = null
-    ): Credential {
-        if (id !in allIds()) {
-            throw IllegalArgumentException("Can't replace non-existing token with id: $id")
-        }
-        val credential = credentialsCache[id] ?: Credential(oidcClient, this, id, token, tags ?: emptyMap())
-        credential.storeToken(token, security, tags, isDefault)
-        credentialsCache[id] = credential
-        return credential
-    }
-
-    internal suspend fun internalReplaceCredential(
-        id: String,
-        token: Token,
-        tags: Map<String, String>,
-        security: Credential.Security?,
-        isDefault: Boolean?
-    ) {
-        val idToken = token.idToken?.let { jwtParser.parse(it) }
-        val payloadData = idToken?.deserializeClaims(JsonObject.serializer())
-        val previousMetadata = metadata(id) ?: throw IllegalStateException("Token metadata for $id not found")
-        val newMetadata = previousMetadata.copy(
-            tags = tags,
-            payloadData = payloadData,
-            isDefault = isDefault ?: previousMetadata.isDefault
-        )
-        storage.replace(id, token, newMetadata, security)
-    }
+    suspend fun replaceToken(id: String, token: Token) = storage.replace(id, token)
```

Comment:
A lot of the logic in replace earlier was for handling if the user called CredentialDataSource.replace directly, or if it was called from a Credential object. This is all that's needed now since it's always called internally.